src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc

```diff
@@ -720,208 +636,123 @@ void OpticalTactilePluginPrivate::DepthCameraCallback(
+    this->extendedSensing)
+    && (_point.Y() >= - this->sensorSize.Y() / 2 -
+    this->extendedSensing);
+
+  bool insideZ = (_point.Z() <= this->sensorSize.Z() / 2 +
+    this->extendedSensing)
+    && (_point.Z() >= - this->sensorSize.Z() / 2 -
+    this->extendedSensing);
+
+  return (insideX && insideY && insideZ);
 }
 
 //////////////////////////////////////////////////
 void OpticalTactilePluginPrivate::ComputeNormalForces(
-                    const ignition::msgs::PointCloudPacked &_msg,
-                    const bool _visualizeForces)
+  const ignition::msgs::PointCloudPacked &_msg,
+  const bool _visualizeForces)
 {
-  // 2D vector to store normal force vectors
-  std::vector<std::vector<ignition::math::Vector3f>> normalForces(_msg.width(),
-      std::vector<ignition::math::Vector3f>(_msg.height()));
-
-  // Visualization is downsampled according to plugin parameter <resolution>
-  // if <visualize_forces> is set to true
-  bool visualizeRow = true;
-  int rowCounter = 0;
+  // Get data from the message
+  std::string rawMsgBuffer = _msg.data();
+  this->msgBuffer = rawMsgBuffer.data();
 
-  bool visualizeColumn = true;
-  int columnCounter = 0;
+  // Declare variables for storing the XYZ points
+  ignition::math::Vector3f p1, p2, p3, p4, markerPosition;
 
-  // Variable for setting the markers id through the iteration
-  int marker_id = 1;
+  // todo(anyone) Improve performance once the following issue is completed:
+  // https://github.com/ignitionrobotics/ign-math/issues/144
 
+  // We don't get the image's edges because there are no adjacent points to
+  // compute the forces
   for (uint64_t j = 1; j < (_msg.height() - 1); ++j)
   {
     for (uint64_t i = 1; i < (_msg.width() - 1); ++i)
     {
-      // Compute normal forces
-      double dxdi =
-          (this->imageXYZ[i+1][j][0] -
-          this->imageXYZ[i-1][j][0])/
-          std::abs(this->imageXYZ[i+1][j][1] -
-          this->imageXYZ[i-1][j][1]);
-
-      double dxdj =
-          (this->imageXYZ[i][j+1][0] -
-          this->imageXYZ[i][j-1][0])/
-          std::abs(this->imageXYZ[i][j+1][2] -
-          this->imageXYZ[i][j-1][2]);
+      // Get points for computing normal forces
+      p1 = this->MapPointCloudData(i + 1, j);
+      p2 = this->MapPointCloudData(i - 1, j);
+      p3 = this->MapPointCloudData(i, j + 1);
+      p4 = this->MapPointCloudData(i, j - 1);
+
+      float dxdi = (p1.X() - p2.X()) / std::abs(p1.Y() - p2.Y());
+      float dxdj =  (p3.X() - p4.X()) / std::abs(p3.Z() - p4.Z());
 
       ignition::math::Vector3f direction(-1, -dxdi, -dxdj);
 
       // todo(anyone) multiply vector by contact forces info
-      normalForces[i][j] = direction.Normalized();
+      ignition::math::Vector3f normalForce = direction.Normalized();
 
-      // todo(mcres) normal forces will be published even if visualization
-      // is turned off
+      // todo(mcres) Normal forces will be computed even if visualization
```

Comment:
Replace `will be` with something more clear. `should be` (i.e. and are not)? Or `are`?

Reply:
e216bb2